### PR TITLE
add minetest server

### PR DESCRIPTION
**Pull request by mikey0000**
> To ease and accelerate the PR submission, please use the template below to provide all requested information.
> You can find guidelines on which docker image to use in the [Rockstor's documentation](http://rockstor.com/docs/contribute_rockons.html#which-docker-image-should-i-use), 
> as well as examples of proper formatting in other rock-ons ([here](https://github.com/rockstor/rockon-registry/blob/master/teamspeak3.json)
> and [here](https://github.com/rockstor/rockon-registry/blob/master/nextcloud-official.json))

Fixes # .

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: Minetest
- website: minetest.net
- description: Rockon for minetest server

### Information on docker image
- docker image: r/linuxserver/minetest
- is an official docker image available for this project?: 
using the linuxserver.io image

### Checklist
- [x] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [x] `"description"` object lists and links to the docker image used
- [x] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [x] `"website"` object links to project's main website

### CODE REVIEW

**Comment by mikey0000** [comment]
Hi guys, please feel free to make suggestions on a better description, I might try for a better one tomorrow.

**Comment by mikey0000** [comment]
Is there a way to set args as an optional instead of required?

**Comment by phillxnet** [comment]
@mikey0000  Hello again, and thanks for another Rock-on submission.

Can't look too closely at this pr right now but they appear to have an ssl cert issue with their site:
https://minetest.net/
We should probably reach out to them on that front.

> Is there a way to set args as an optional instead of required?

Don't think so, at least yet, but @FroggyFlox should be able to confirm or deny that one as they have been working with that part of the code more recently. We do have an optional "devices" section now however so there is precedent.

**Comment by mikey0000** [comment]
> @mikey0000 Hello again, and thanks for another Rock-on submission.
> 
> Can't look too closely at this pr right now but they appear to have an ssl cert issue with their site:
> https://minetest.net/
> We should probably reach out to them on that front.
> 
> > Is there a way to set args as an optional instead of required?
> 
> Don't think so, at least yet, but @FroggyFlox should be able to confirm or deny that one as they have been working with that part of the code more recently. We do have an optional "devices" section now however so there is precedent.

Cool, yeah looks like they actually don't have https on the main website, did https by default as their wiki supports it.

**Comment by FroggyFlox** [comment]
Hi @mikey0000 ,

Sorry for not getting back with you earlier than that.

>>     Is there a way to set args as an optional instead of required?
> Don't think so, at least yet, but @FroggyFlox should be able to confirm or deny that one as they have been working with that part of the code more recently. We do have an optional "devices" section now however so there is precedent.

We indeed cannot do that at the moment, but we do have it planned in the not-so-distant future. We do have a corresponding issue to track it:
https://github.com/rockstor/rockstor-core/issues/2094#issue-530511707

For the time being, I'm tempted to recommend removing the `CLI_ARGS` env because of it, and we can always put it back once optional environment variables are possible. I couldn't test whether or not the minetest instance was indeed reachable, but from the logs, it seems to start without error. This is, of course, without the `CLI_ARGS` defined.

What do you think about removing it? Does it actually remove important features? As they are defined as optional on the docker hub documentation, I tend to believe we can do without them at the moment but I'm not a Minetest user so I can't know for sure. Is there any setting that can only be defined through the `CLI_ARGS` environment variable and not through the conf files, for instance?

**Comment by mikey0000** [comment]
Yeah it can be set in the conf so can remove it.

14/06/2020 12:58:28 am FroggyFlox <notifications@github.com>:

**Comment by mikey0000** [comment]
@FroggyFlox I finally removed the cli args :smiley_cat:

File: minetest.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                        "label": "UID to run minetest as.",
+                        "index": 1
+                    },
+                    "PGID": {
+                        "description": "Enter a valid GID to use along with the above UID. It(or the above UID) must have full permissions to the Share mapped in the previous step.",
+                        "label": "GID to run minetest as.",
+                        "index": 2
+                    },
+                    "TZ": {
+                        "description": "Enter timezone e.g Europe/London these can be found ...",
```
Comment (FroggyFlox) [inline minetest.json:34]: Minor edit:
```suggestion
                        "description": "Enter timezone e.g Europe/London (see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).",
```

**Comment by FroggyFlox** [comment]
Hi @mikey0000 ,

First of all, sorry again for such a delay in getting to your submission... 
I've now re-tested and it still works as intended, thanks a lot for your original submission and your continued time and effort in making the revisions!

As you can see in the review, I only noticed a minor improvement that we can make in the description of the `TZ` environment variable. I believe we can merge your PR once that minor edit is included.

Sorry again for such a delay in getting to this.

**Comment by mikey0000** [comment]
Ah sorry must have missed that one, will address it. No worries we all have lives outside github 😁
26/08/2020 1:14:50 am FroggyFlox <notifications@github.com>:

**Comment by mikey0000** [comment]
Hey guys life happens and I completely forgot about finishing this. Will address right now.

**Review by FroggyFlox — CHANGES_REQUESTED** [review]
Hi @mikey0000 ,

Thanks a lot for following-up and making the changes! With all the time it took me to review it, I unfortunately have another small suggestion that I would like to see included. As you can see, it only relates to an harmonization of the `"description"` field with other rock-ons that we now try to follow.

Everything else still works as intended so I'll merge your PR once we have that little addition to the `"description"` field in.

Thanks again!

**Comment by mikey0000** [comment]
@FroggyFlox I think I addressed the description?

File: minetest.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    },
+                    "TZ": {
+                        "description": "Enter timezone e.g Europe/London (see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).",
+                        "label": "Timezone to run minetest server in.",
+                        "index": 3
+                    }
+                }
+            }
+        },
+        "description": "An open source voxel game engine clone of the popular Minecraft, see <em>more info</em> for details.</p><p>Based on the docker image by <a href='https://www.linuxserver.io/' target='_blank'>Linuxserver.io</a>: <a href='https://hub.docker.com/r/linuxserver/minetest' target='_blank'>https://hub.docker.com/r/linuxserver/minetest</a>, available for amd64, arm32v7 and arm64v8.</p>",
```
Comment (phillxnet) [inline minetest.json:41]:
```suggestion
        "description": "An open source voxel game engine clone of the popular Minecraft, see <em>more info</em> for details.</p><p>Based on the docker image by <a href='https://www.linuxserver.io/' target='_blank'>Linuxserver.io</a>: <a href='https://hub.docker.com/r/linuxserver/minetest' target='_blank'>https://hub.docker.com/r/linuxserver/minetest</a>, available for amd64 and arm64 architecture.</p>",
```

Comment (phillxnet) [reply minetest.json]: @mikey0000 Sorry about this but we have, in the interim in pr #310 recently standardised on our architecture wording for all existing Rock-ons. I know it seems trivial but we are trying to move, step-by-step, too these elements having their own tag and doing this should help with that longer term goal.
Hence my suggestion here.

Thanks for picking-up this pull request again. Nearly there now I think, judging from a quick read of @FroggyFlox's past comments.

**Comment by phillxnet** [comment]
@mikey0000 What do you know, we finally got this one sorted.

Adding as initially reviewed by @FroggyFlox with a light install test by myself:
```
[03/Feb/2023 15:28:41] DEBUG [system.osi:208] Running command: /usr/bin/docker pull linuxserver/minetest:latest
[03/Feb/2023 15:28:49] DEBUG [system.osi:208] Running command: /usr/bin/docker run -d --restart=unless-stopped --name minetest -v /mnt2/minetest-config:/config/.minetest -v /etc/localtime:/etc/localtime:ro -p 30000:30000/udp -e PUID=1010 -e TZ=Europe/Lisbon -e PGID=100 linuxserver/minetest:latest
[03/Feb/2023 15:28:50] DEBUG [storageadmin.views.rockon_helpers:210] Set rock-on 91 state to installed
[03/Feb/2023 15:28:50] INFO [storageadmin.tasks:64] Task [install], id: 22fcc389-179d-4d97-a182-efe31c6cc122 completed OK
```

Thanks all around for the super long persistent tail on this one. Lets just get it in, and out, to get some field testing on it.

**Review by phillxnet — APPROVED** [review]
As per comments, this super long tail has likely come to an end.
I'll pop in as it looks like all requested changes have been applied.

**Comment by phillxnet** [comment]
minetest-installed

**Comment by mikey0000** [comment]
🎉 thanks @phillxnet, nice to see it finally make it.

**Comment by phillxnet** [comment]
@mikey0000 Yes, it's great to get Rock-ons finally through review. Always tweaks that can be made however so bit of a balancing act.

I've just now published the recent changes in this repository to production so this and a number of other Rock-ons should now show up after an "Update" button push in the Rock-ons page (All tab).

Thanks again for hanging in there.

**Pull request merged**